### ISSUE TITLE
Notify developer(s) when we approve versions while resolving a cinderjob

### ISSUE THREAD

**Issue by eviljeff** [mozilla/addons-server#21826]
As part of the changes to implement abuse report handling in the reviewer tools, if a job (containing abuse reports) is resolved with a reviewer tools action we skip the existing reviewer tools logic for sending emails, and instead rely on the CinderJob/CinderAction methods to send the correct emails.  This means in cases where an abuse report is judged to be not valid we don't notify the developer.

This is generally fine, except for the case when a version is being approved (i.e. it is not already approved, because it hasn't been auto-approved) where we _do_ need to notify the developer so they know it's signed and available for download, but for abuse report purposes we don't notify them.  

When https://mozilla-hub.atlassian.net/browse/AMO-116 is completed we will have reworked all the emails anyway, so what is needed for now is a temporary fix.

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/ADDSRV-696)

**Comment by ioanarusiczki** [comment]
@eviljeff 

I tried with a listed extension and a static theme. When I have a new version awaiting review and there's a previous version reported for [abuse within the add-on](https://reviewers.addons.allizom.org/en-US/reviewers/review/1022613) or it's [escalated to AMO ](https://reviewers.addons.allizom.org/en-US/reviewers/review/1022011). I checked the report from Outstanding DSA related reports to resolve:

![Approve and report](https://github.com/mozilla/addons-server/assets/33448286/603008c4-abc9-4ad8-b2d3-0328edead89b)

2 emails are sent: one to the reporter notifying that content does not violate Mozilla's policies and an email sent to the developer about the version being approved.
